myriad-scheduler/src/main/java/org/apache/myriad/MyriadModule.java

```diff
@@ -103,11 +103,12 @@ protected void configure() {
 
     MapBinder<String, TaskFactory> mapBinder = MapBinder.newMapBinder(binder(), String.class, TaskFactory.class);
     mapBinder.addBinding(NodeManagerConfiguration.NM_TASK_PREFIX).to(NMTaskFactoryImpl.class).in(Scopes.SINGLETON);
-    Map<String, ServiceConfiguration> auxServicesConfigs = cfg.getServiceConfigurations().orNull();
-    if (auxServicesConfigs != null) {
+
+    if (cfg.getServiceConfigurations().isPresent()) {
+      Map<String, ServiceConfiguration> auxServicesConfigs = cfg.getServiceConfigurations().get();
       for (Map.Entry<String, ServiceConfiguration> entry : auxServicesConfigs.entrySet()) {
-        String taskFactoryClass = entry.getValue().getTaskFactoryImplName().orNull();
```

Comment:
I think I mentioned this before but an elsewhere (config maybe?) Map would have the same effect.  It's pretty common to pass an empty container vs an optional of a container class.